components/ai_chat/core/browser/associated_content_driver.h

```diff
@@ -0,0 +1,167 @@
+  void OnPageContentUpdated(std::string content,
+                            bool is_video,
+                            std::string invalidation_token);
+
+  // To be called when a page navigation is detected and a new conversation
+  // is expected.
+  void OnNewPage(int64_t navigation_id) override;
+
+  // Begin the alternative content fetching (print preview / OCR) by
+  // sending a message to an observer with access to the layer this can
+  // be performed on (i.e. the UI layer can use Print Preview).
+  // Since this AssociatedContentDriver might be open in multiple WebUIs,
+  // we can no longer rely on the WebUI class observing its conversation 1:1.
+  // We might end up with multiple UIs performing the print preview fetching
+  // for a single conversation. Instead we need some single entity that is
+  // able to perform this work, either by relocating AIChatTabHelper to
+  // brave/browser/ui, or having a singleton/KeyedService observe this event
+  // void NotifyPrintPreviewRequested(bool is_pdf);
+ private:
+  friend class ::AIChatUIBrowserTest;
+  FRIEND_TEST_ALL_PREFIXES(::AIChatUIBrowserTest, PrintPreviewFallback);
+  FRIEND_TEST_ALL_PREFIXES(AssociatedContentDriverUnitTest,
+                           ParseSearchQuerySummaryResponse);
+
+  void OnGeneratePageContentComplete(int64_t navigation_id,
+                                     std::string contents_text,
+                                     bool is_video,
+                                     std::string invalidation_token);
+  void OnExistingGeneratePageContentComplete(
+      ConversationHandler::GetPageContentCallback callback,
+      int64_t navigation_id);
+
+  void OnSearchSummarizerKeyFetched(
+      ConversationHandler::GetStagedEntriesCallback callback,
+      int64_t navigation_id,
+      const std::optional<std::string>& key);
+  void OnSearchQuerySummaryFetched(
+      ConversationHandler::GetStagedEntriesCallback callback,
+      int64_t navigation_id,
+      api_request_helper::APIRequestResult result);
+  static std::optional<SearchQuerySummary> ParseSearchQuerySummaryResponse(
+      const base::Value& value);
+
+  raw_ptr<PrefService> pref_service_;
+  raw_ptr<AIChatMetrics> ai_chat_metrics_;
+  std::unique_ptr<AIChatCredentialManager> credential_manager_;
+
+  scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory_;
+
+  // Used for fetching search query summary.
+  std::unique_ptr<api_request_helper::APIRequestHelper> api_request_helper_;
+
+  base::ObserverList<Observer> observers_;
+  std::unique_ptr<base::OneShotEvent> on_page_text_fetch_complete_;
+  bool is_page_text_fetch_in_progress_ = false;
+  std::string cached_text_content_;
+  std::string content_invalidation_token_;
+  bool is_video_ = false;
+
+  // Handlers that are interested in this content for the current navigation.
+  std::set<raw_ptr<ConversationHandler>> associated_conversations_;
+
+  // Store the unique ID for each "page" so that
```

Comment:
This is just for the current page, right?
```suggestion
  // Store the unique ID for the current "page" so that
```

Reply:
Yes, thanks - this comment was copied from the original `ConversationDriver` which handled multiple pages, although I guess your wording would have been better then too